python/cudf/cudf/core/column/datetime.py

```diff
@@ -229,6 +229,27 @@ def find_last_value(self, value, closest=False):
     def is_unique(self):
         return self.as_numerical.is_unique
 
+    def can_cast_safely(self, to_dtype):
+        if np.issubdtype(to_dtype, np.datetime64):
+
+            to_res = to_dtype.name[-3:-1]
+            self_res = self.dtype.name[-3:-1]
+
+            epoch = np.datetime64(0, to_res)
+            max_int = np.iinfo(np.dtype("int64")).max
+
+            max_dist = self.max() - epoch.astype(self.dtype)
+            min_dist = self.min() - epoch.astype(self.dtype)
+
+            if max_dist <= np.timedelta64(max_int, to_res).astype(
+                "m8[" + self_res + "]"
+            ) and min_dist <= np.timedelta64(max_int, to_res).astype(
+                "m8[" + self_res + "]"
+            ):
+                return True
+            else:
+                return False
+
```

Comment:
This function returns `None` if the condition on line 233 evaluates to `False` :)

Reply:
Nice catch - fixed.